reference/implementing-a-concept-exercise.md

```diff
@@ -0,0 +1,83 @@
+# How to Implement a C Concept Exercise
+
+This document describes how to implement a concept exercise for the C track.
+
+**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read the following documents:
+
+- [The features of v3][docs-features-of-v3].
+- [Rationale for v3][docs-rationale-for-v3].
+- [What are concept exercise and how they are structured?][docs-concept-exercises]
+
+Please also watch the following video:
+
+- [The Anatomy of a Concept Exercise][anatomy-of-a-concept-exercise].
+
+As this document is generic, the following placeholders are used:
+
+- `<SLUG>`: the slug of the exercise in kebab-case (e.g. `calculator-conundrum`).
+- `<NAME>`: the name of the exercise in snake_case (e.g. `calculator_conundrum`).
+- `<CONCEPT_SLUG>`: the slug of one of the exercise's concepts in kebab-case (e.g. `anonymous-methods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue.
+
+To implement a concept exercise, the following files must be added:
+
+<pre>
+languages
+└── c
+    ├── concepts
+    |   └── &lt;CONCEPT_SLUG&gt;
+    |       ├── about.md
+    |       └── links.json
+    └── exercises
+        └── concept
+            └── &lt;SLUG&gt;
+                ├── .docs
+                │   ├── instructions.md
+                │   ├── introduction.md
+                │   ├── hints.md
+                │   └── source.md (required if there are third-party sources)
+                ├── .meta
+                │   ├── config.json
+                │   ├── design.md
+                │   └── exemplar.c
+                ├── makefile
+                ├── &lt;NAME&gt;.c
+                ├── &lt;NAME&gt;.h
+                ├── test_&lt;NAME&gt;.c
+                └── test-framework
+</pre>
+
+## Step 1: Add code files
+
+The code files are track-specific and should be designed to help the student learn the exercise's concepts. The following files must be added (not necessarily in this order):
+
+- `makefile`: the makefile for the stub files and tests
+- `<NAME>{.c|.h}`: the stub source and header implementation files, which are the starting point for students to work on the exercise.
+- `test_<NAME>.c`: the unit test source file.
+- `test-framework`: this directory contains the source files of unit test framework, [Unity][unity]
```

Comment:
We probably need to revisit discussion on this structure and the requirements for it, to know the correct approach. 
If practice exercises are to follow the same structure, then a lot of the file renames in this commit are wrong in that they do not result in this structure.

Reply:
This comes about because the v3 track for C was started, but then I didnt have time to keep up with keeping it up to date with the evolving plans for v3 - and because our existing structure is different.

@ErikSchierboom is it intended that this PR result in a conformant repo, or just that it cover what is described in your comment and that we fix broken things afterwards?

Reply:
The latter. Tracks should preferrably merge this PR when they can and then fix things later.